### PR TITLE
CSAT banner for “Last scan date”  (MNTOR-3146)

### PR DESCRIPTION
**Pull request by flozia**
# References:

Jira: [MNTOR-3146](https://mozilla-hub.atlassian.net/browse/MNTOR-3146)

# Description

This PR adds the CSAT banner for users in the “Last scan date” experiment. This change comes with a refactor of how we add and handle multiple CSAT surveys.

# How to test

(Re)build the Nimbus experiments file with `npm run build-nimbus`.

**Storybook**
The CSAT survey banner can be easily tested in Storybook:
- Local: http://localhost:6006/?path=/story/csatsurvey--csat-survey-latest-scan-date
- Netlify: https://deploy-preview-4611--fx-monitor-storybook.netlify.app/?path=/story/csatsurvey--csat-survey-latest-scan-date

Using the controls the props can be changed. If the banner has been dismissed or interacted with the set cookies will prevent it from showing up again and the respective cookie needs to be deleted from the browser’s developer console.

**Monitor (local)**
Visit the dashboard with a free/plus user: http://localhost:6060/user/dashboard. Depending on the dashboard tab the respective CSAT survey should be shown.

The testing instructions for the “Automatic removal CSAT banner” introduced in #4421 are staying the same, except that we don’t need to add the feature flags anymore since it was converted to an experiment.

# Checklist (Definition of Done)

- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

[MNTOR-3146]: https://mozilla-hub.atlassian.net/browse/MNTOR-3146?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: config/nimbus.yaml [hunk 1]
```diff
@@ -33,3 +33,31 @@ features:
...
+        default: false
+    defaults:
+      - channel: local
+        value: { "enabled": true }
+      - channel: staging
+        value: { "enabled": true }
+      - channel: production
+        value: { "enabled": false }
+  latest-scan-date-csat-survey:
+    description: Show the CSAT survey for users that are enrolled in the `latest-scan-date` experiment
```
Comment (rhelmer) [inline config/nimbus.yaml:51]:
```suggestion
  last-scan-date-csat-survey:
    description: Show the CSAT survey for users that are enrolled in the `last-scan-date` experiment
```

The experiment is named `last-scan-date`, do we want it to be the same here for consistency?

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx [hunk 2]
```diff
@@ -3461,4 +3529,74 @@ describe("CSAT survey banner", () => {
...
+        elapsedTimeInDaysSinceInitialScan={1}
+        experimentData={{
+          ...defaultExperimentData,
+          "automatic-removal-csat-survey": {
+            enabled: true,
+          },
+          "last-scan-date": {
+            enabled: true,
+          },
+          "latest-scan-date-csat-survey": {
```
Comment (rhelmer) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx:3551]:
```suggestion
          "last-scan-date-csat-survey": {
```

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx [hunk 3]
```diff
@@ -3461,4 +3529,74 @@ describe("CSAT survey banner", () => {
...
```
Comment (rhelmer) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx:3592]:
```suggestion
          "last-scan-date-csat-survey": {
```

File: src/app/components/client/csat_survey/surveys/latestScanDateCsatSurvey.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import {
+  CsatSurveyProps,
+  RelevantSurvey,
+  SurveyData,
+  getRelevantSurveys,
+} from "./csatSurvey";
+
+const surveyData: SurveyData = {
+  id: "csat_survey_latest_scan_date",
+  requiredExperimentIds: ["last-scan-date", "latest-scan-date-csat-survey"],
```
Comment (rhelmer) [inline src/app/components/client/csat_survey/surveys/latestScanDateCsatSurvey.ts:14]:
```suggestion
  requiredExperimentIds: ["last-scan-date", "last-scan-date-csat-survey"],
```

File: src/app/components/client/stories/CsatSurvey.stories.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  name: "LatestScanDate",
+  args: {
+    activeTab: "fixed",
+    user: createUserWithPremiumSubscription(),
+    experimentData: {
+      ...defaultExperimentData,
+      "last-scan-date": {
+        enabled: true,
+      },
+      "latest-scan-date-csat-survey": {
```
Comment (rhelmer) [inline src/app/components/client/stories/CsatSurvey.stories.ts:44]:
```suggestion
      "last-scan-date-csat-survey": {
```

**Review by rhelmer — CHANGES_REQUESTED** [review]

File: config/nimbus.yaml (hunk 1 above)
Comment (flozia) [reply config/nimbus.yaml]: Yes a 💯% — renamed in [306872c](https://github.com/mozilla/blurts-server/pull/4611/commits/306872c977ead1a9cc7c694ad667811fbe99c073).

**Review by rhelmer — APPROVED** [review]

**Review by jespy2 — APPROVED** [review]
LGTM

**Pull request merged**